### ISSUE TITLE
Document supported hardware architectures

### ISSUE THREAD

**Issue by zamazan4ik** [khonsulabs/bonsaidb#258]
Hi!

Could you please put the information about supported architectures to the documentation please? E.e. about supported architectures for different operating systems, some specific requirements to the supported instructions, if you have any (e.g. maybe AVX is required - I do not know).

This kind of information is important for the end-users.

Thanks in advance!

**Comment by ecton** [comment]
Thank you for the feedback. I agree that this is important, but I'm also not sure what to document at this time. As far as I'm aware, BonsaiDb will run wherever the standard library compiles to. There may be specific dependencies of certain server-related functionality that I'm just not aware have extra requirements, but I know people have had success compiling for a variety of architectures.

Are you aware of any platforms that the standard library supports that BonsaiDb currently fails to compile for or work on? I would love to either document the issues or fix them.

**Comment by zamazan4ik** [comment]
No, I am not aware about such platforms. Maybe you can at least write a note something like "The library should work on any where the standard library works". Also, maybe you have platforms, which are more battle-tested than others (probably Linux/x86-64 or anything else) - you can write about this too, since it can at least show that on these platforms a chance to find a bug should be lower.
